forward_script/src/QueryBuilder.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+
+namespace Wikimedia\ForwardScript;
+
+class QueryBuilder {
+
+	public function getQuery( $info, $pageName, $id="", $lat="", $lon="" ) {
+		$category = preg_replace( "/^(:?Category|Kategorie):/", "", $info->category );
+		$query = [ "categories" => $category ];
+		if ( $lat ) {
+			$query["lat"] = $lat;
+		}
+		if ( $lon ) {
+			$query["lon"] = $lon;
+		}
+		if ( $id && empty( $info->duplicate_ids ) && empty( $info->id_not_found ) ) {
+			$query["objref"] = implode( "|", ["de", $pageName, $id] );
+		}
+		if ( !empty( $info->valid_id ) ) {
+			$query["fields[]"] = $id;
+		}
+		$queryString = "";
+		foreach ( $query as $param => $value ) {
+			$queryString .= "&" . $param . "=" . urlencode( $value );
+		}
+		return $queryString;
```

Comment:
This method could be simplified by using [`http_build_query`](http://php.net/manual/de/function.http-build-query.php). Additionally we could avoid the three if-statements by passing it an array `['id' => ..., 'lat' => ..., 'lon' => ...]` instead of the 3 arguments and `array_filter`ing out falsy values.